### ISSUE TITLE
Tiller image installed not latest

### ISSUE THREAD

**Issue by magic7s** [hashicorp/terraform-provider-helm#118]
Provider is installing not latest version, causing errors when running other helm commands.
```
macbook$ helm reset
Error: incompatible versions client[v2.10.0] server[v2.9.0]
```
Version:
```
Client: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.0", GitCommit:"f6025bb9ee7daf9fee0026541c90a6f557a3e0bc", GitTreeState:"clean"}
```

Static defined version:
https://github.com/mcuadros/terraform-provider-helm/blob/master/helm/provider.go#L64

### Terraform Version
Terraform v0.11.8
+ provider.aws v1.31.0
+ provider.external v1.0.0
+ provider.google v1.16.2
+ provider.helm (unversioned)
+ provider.local v1.1.0
+ provider.null v1.0.0
+ provider.template v1.0.0

### Terraform Configuration Files
```hcl
resource "null_resource" "helm-init" {
  # Changes to any instance of the cluster requires re-provisioning
  triggers = {
    cluster_instance_cert = "${ var.kube_cluster_cert }"
  }
  
  provisioner "local-exec" {
    environment = { KUBECONFIG = "${ var.kubeconfig_file }" }
    command = "helm init"
  }
  provisioner "local-exec" {
    when    = "destroy"
    environment = { KUBECONFIG = "${ var.kubeconfig_file }"}
    command = "helm reset"
  }
  depends_on = ["null_resource.istio-svc-act"]
}
```

### Expected Behavior
Should match shipping version of helm client

### Actual Behavior
Tiller version installed not latest.

**Comment by pdecat** [comment]
The default tiller version was updated to v2.14.0 in order to match the embedded helm client in #277.

This was released with [v0.10.0 of the helm provider](https://github.com/terraform-providers/terraform-provider-helm/blob/master/CHANGELOG.md#0100-june-18-2019).

Upgrading an already installed tiller deployment is not supported by the provider right now: #109

**Comment by rporres** [comment]
Tiller image can be set on provider via `tiller_image` option. The default, as @pdecat comments, should match the helm client
